src/misc/setup.cpp

```diff
@@ -112,49 +120,36 @@ bool Value::operator==(Value const& other) const {
 			if((*_string) == (*other._string)) return true;
 			break;
 		default:
-			E_Exit("comparing stuff that doesn't make sense");
-			break;
+		        LOG_ERR("SETUP: Comparing stuff that doesn't make sense");
+		        break;
 	}
 	return false;
 }
-bool Value::SetValue(string const& in,Etype _type) {
-	/* Throw exception if the current type isn't the wanted type 
-	 * Unless the wanted type is current.
-	 */
-	if(_type == V_CURRENT && type == V_NONE) throw WrongType();
-	if(_type != V_CURRENT) { 
-		if(type != V_NONE && type != _type) throw WrongType();
-		type = _type;
-	}
-	bool retval = true;
-	switch(type){
-		case V_HEX:
-			retval = set_hex(in);
-			break;
-		case V_INT:
-			retval = set_int(in);
-			break;
-		case V_BOOL:
-			retval = set_bool(in);
-			break;
-		case V_STRING:
-			set_string(in);
-			break;
-		case V_DOUBLE:
-			retval = set_double(in);
-			break;
+bool Value::SetValue(const std::string &in, const Etype _type)
+{
+	assert(type == V_NONE || type == _type);
+	type = _type;
 
-		case V_NONE:
-		case V_CURRENT:
-		default:
-			/* Shouldn't happen!/Unhandled */
-			throw WrongType();
-			break;
+	bool retval = true;
+	switch (type) {
+	case V_HEX: retval = set_hex(in); break;
+	case V_INT: retval = set_int(in); break;
+	case V_BOOL: retval = set_bool(in); break;
+	case V_STRING: set_string(in); break;
+	case V_DOUBLE: retval = set_double(in); break;
+
+	case V_NONE:
+	case V_CURRENT:
+	default:
+		LOG_ERR("SETUP: Unhandled type when setting value: %s", in.c_str());
+		retval = false;
+		break;
 	}
 	return retval;
 }
 
-bool Value::set_hex(std::string const& in) {
+bool Value::set_hex(const std::string &in)
```

Comment:
Maybe could change to UpperCamelCase if these aren't exported.

Reply:
Good catch.